src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageElementDescriptionProvider.java

```diff
@@ -23,10 +26,17 @@
  */
 public class LSPUsageElementDescriptionProvider implements ElementDescriptionProvider {
     @Override
-    public @Nullable @NlsSafe String getElementDescription(@NotNull PsiElement element, @NotNull ElementDescriptionLocation location) {
+    @Nullable
+    @NlsSafe
+    public String getElementDescription(@NotNull PsiElement element, @NotNull ElementDescriptionLocation location) {
         if (element instanceof LSPUsageTriggeredPsiElement) {
             return LanguageServerBundle.message("usage.description");
         }
+        // If this is for Ctrl/Cmd+Mouse hover, try to get the element description from the documentation provider

```

Comment:
And another branch that's needed specifically for the **Ctrl/Cmd+Hover** use case.